app.js

```diff
@@ -1499,6 +1499,162 @@ app.delete("/room/:roomId", function (req, res) {
     });
 });
 
+
+//suggest meeting times given list of participants
+app.post("/api/meetingSuggestion", function (req, res) {
+
+    //currently only look 3 days ahead, and assume 7AM - 5PM workday. only search on weekdays
+
+    if(!req.body.participants) {
+      res.statusCode = 400;
+      console.log();
+      res.json({
+        "requestURL":  "/room",
+        "action": "post",
+        "status": 400,
+        "message": "Bad Request",
+        "timestamp": new Date()
+      });
+    }
+
+    else {
+
+      var participants = req.body.participants || null;
+
+      //using ADDDATE(CURDATE(), 4) just to limit the amount of meetings to search through later
+      var getMeetingSql = "SELECT start_datetime, end_datetime FROM ebdb.meeting WHERE " //Where end_datetime < ADDDATE(CURDATE(), 4) AND"
+                         + "calendar IN (SELECT primary_calendar FROM ebdb.user WHERE email IN (?));";
+      var inserts = [participants];
+
+      pool.query(getMeetingSql, inserts, function(error, results, fields) {
+        if(error) {
+          res.statusCode = 500;
+          console.log(error);
+          res.json({
+            "requestURL":  "/meetingSuggestion",
+            "action": "post",
+            "status": 500,
+            "message": "Query failed for calendar",
+            "timestamp": new Date()
+          });
+        }
+        else {
+          
+          var meetings = results;
+
+          //construct time table
+          var currDate = new Date();
+          //console.log(currDate.toISOString());
+
+          var currHour = currDate.getHours();
+
+          var currMin = currDate.getMinutes();
+
+          currDate.setSeconds(0);
+          currDate.setMilliseconds(0);
+
+          //round up to the nearest 30 min
+          if(currMin > 30) {
+            currMin = 0;
+            currHour++;
+            currDate.setHours(currHour);
+            currDate.setMinutes(currMin);
+
+          }
+          else {
+            currMin = 30;
+            currDate.setHours(currHour);
+            currDate.setMinutes(currMin);
+          }
+
+          //if weekend, or friday after 5, start searching monday at 7
+          if(currDate.getDay() == 6 || currDate.getDay == 0 ||
+             (currDate.getDay() == 5 && currDate.getHour >= 5)) {
+            if(currDate.getDay() == 6) {
+              currDate.setDate(currDate.getDate() + 2);
+            }
+            else if(currDate.getDay() === 0) {
+              currDate.setDate(currDate.getDate() + 1);
+            }
+            else {
+              currDate.setDate(currDate.getDate() + 3);
+            }
+            currHour = 7;
+            currMin = 0;
+            currDate.setHours(7);
+            currDate.setMinutes(0);
+          }
+
+          var timetable = {};
+          //search for the next 3 days
+          for(var i = 0; i < 48; i++) {
```

Comment:
It's a bit hard to read with the commented out code.  I would suggest cleaning the code up by removing the commented out code and refactoring into separate functions where possible.

For example, code that creates and populates the timetable could be refactored into its own function which takes a list of meetings, days to search, and start and stop times as parameters and returns the populated timetable.

Reply:
by "days to search" do you mean number of days ahead to look (3), or like a list of workdays?